h2o-py/h2o/frame.py

```diff
@@ -3182,9 +3182,16 @@ def cor(self, y=None, na_rm=False, use=None, method="Pearson"):
         assert_is_type(y, H2OFrame, None)
         assert_is_type(na_rm, bool)
         assert_is_type(use, None, "everything", "all.obs", "complete.obs")
-        if y is None:
-            y = self
-        if use is None: use = "complete.obs" if na_rm else "everything"
+        y_categorical = any(self.types[col_name] == "enum" for col_name in y)
+
+        if y_categorical:
+            num_unique_levels = {col: len(self[col].levels()) for col in y}
```

Comment:
Instead of `len(self[col].levels())` use `self[col].nlevels()[0]`. `nlevels` returns the just a list of cardinalities so there is less communication with the backend and lower memory use in the python client. Also since the `y` is an H2OFrame (see the assert on the line 3182) you can use something like `dict(zip(y.columns, y.nlevels()))` to get the same thing.